api/python/quilt3/packages.py

```diff
@@ -1065,7 +1072,15 @@ def dump(self, writable_file):
         return self._dump(writable_file)
 
     def _dump(self, writable_file):
-        writer = jsonlines.Writer(writable_file)
+        json_encode = json.JSONEncoder(ensure_ascii=False).encode
+
+        def dumps(obj):
+            data = json_encode(obj).encode()
+            if len(data) > MAX_MANIFEST_RECORD_SIZE:
```

Comment:
Maybe we should set this limit on the size of user metadata, but then we need to be sure that we have enough space for our own data.

Reply:
I think you are doing it right: we should measure the TOTAL size of the row because that's what S3 select will care about.